### ISSUE TITLE
Can't extend one custom theme from another

### ISSUE THREAD

**Issue by waywardmonkeys** [sphinx-doc/sphinx#1792]
I'm trying to do something that appears to be described in some other issues (like #1347, #1335, #511).

I have a `_themes` directory with 2 things in it:  `opendylan-docs` and `sphinx_rtd_theme`.

In `opendylan-docs`, the theme.conf says that it overrides `sphinx_rtd_theme`.

`opendylan-docs` has a `layout.html` that ``{% extends "sphinx_rtd_theme/layout.html" %} but this results in an error:

```
Exception occurred:
  File "/usr/local/lib/python2.7/site-packages/sphinx/jinja2glue.py", line 159, in get_source
    raise TemplateNotFound(template)
TemplateNotFound: sphinx_rtd_theme/layout.html
The full traceback has been saved in /var/folders/pj/fcfyqt6n2r5cjygxgtnfmv7h0000gn/T/sphinx-err-n7TWFJ.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
```

The contents of that log file are:

```
# Sphinx version: 1.3.1
# Python version: 2.7.9 (CPython)
# Docutils version: 0.12 release
# Jinja2 version: 2.7.3
# Last messages:
#   done
#   building [mo]: targets for 0 po files that are out of date
#   building [html]: targets for 50 source files that are out of date
#   updating environment:
#   0 added, 0 changed, 0 removed
#   looking for now-outdated files...
#   none found
#   preparing documents...
#   done
#   writing output... [  2%] c-ffi/index
# Loaded extensions:
#   alabaster (0.7.3) from /usr/local/lib/python2.7/site-packages/alabaster/__init__.pyc
#   dylandomain.dylandomain (unknown version) from /Users/bruce/Development/dylan/opendylan/documentation/sphinx-extensions/dylandomain/dylandomain.pyc
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/sphinx/cmdline.py", line 245, in main
    app.build(opts.force_all, filenames)
  File "/usr/local/lib/python2.7/site-packages/sphinx/application.py", line 264, in build
    self.builder.build_update()
  File "/usr/local/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 245, in build_update
    'out of date' % len(to_build))
  File "/usr/local/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 316, in build
    self.write(docnames, list(updated_docnames), method)
  File "/usr/local/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 354, in write
    self._write_serial(sorted(docnames), warnings)
  File "/usr/local/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 362, in _write_serial
    self.write_doc(docname, doctree)
  File "/usr/local/lib/python2.7/site-packages/sphinx/builders/html.py", line 445, in write_doc
    self.handle_page(docname, ctx, event_arg=doctree)
  File "/usr/local/lib/python2.7/site-packages/sphinx/builders/html.py", line 786, in handle_page
    output = self.templates.render(templatename, ctx)
  File "/usr/local/lib/python2.7/site-packages/sphinx/jinja2glue.py", line 138, in render
    return self.environment.get_template(template).render(context)
  File "/usr/local/lib/python2.7/site-packages/jinja2/environment.py", line 969, in render
    return self.environment.handle_exception(exc_info, True)
  File "/usr/local/lib/python2.7/site-packages/jinja2/environment.py", line 742, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python2.7/site-packages/sphinx/themes/basic/page.html", line 10, in top-level template code
    {%- extends "layout.html" %}
  File "/Users/bruce/Development/dylan/opendylan/documentation/library-reference/source/../../_themes/opendylan-docs/layout.html", line 3, in top-level template code
    {% extends "sphinx_rtd_theme/layout.html" %}
  File "/usr/local/lib/python2.7/site-packages/sphinx/jinja2glue.py", line 159, in get_source
    raise TemplateNotFound(template)
TemplateNotFound: sphinx_rtd_theme/layout.html
```

**Comment by shimizukawa** [comment]
I could.

`conf.py`:

```
# -*- coding: utf-8 -*-

master_doc = 'index'
project = u'1792'
copyright = u'2015, dev'
author = u'dev'
version = release = '1.3.1'
exclude_patterns = ['_build']

html_theme = 'mytheme'
html_theme_path = ['_themes']
```

`_themes/sphinx_rtd_theme` is copy of sphinx rtd theme.

`_themes/mytheme/theme.conf`:

```
[theme]
inherit = sphinx_rtd_theme
```

`_themes/mytheme/layout.html`:

```
{%- extends "sphinx_rtd_theme/layout.html" %}
```

full project files is here: 
https://dl.dropboxusercontent.com/u/284189/sphinx-issues/issue1792.tgz

Please let me know if it doesn't work fine.

**Comment by waywardmonkeys** [comment]
Your exmaple works ... but I was just debugging this and it was happening because `html_theme_path` is `['../../_themes']` as we have a lot of documents sharing a single theme.

That path value isn't converted to an absolute path in the `Theme` constructor ... and then it is put into the `themepath` as is ... and that results in the `BuiltinTemplateLoader` seeing in the `init` method:

```
theme.get_dirchain():
[u'/Users/bruce/Development/dylan/opendylan/documentation/library-reference/source/../../_themes/opendylan-docs', u'/Users/bruce/Development/dylan/opendylan/documentation/library-reference/source/../../_themes/sphinx_rtd_theme', '/usr/local/lib/python2.7/site-packages/sphinx/themes/basic']

theme.themepath
['../../_themes', '/usr/local/lib/python2.7/site-packages/sphinx/themes']
```

And then when it is searching (in `SphinxFileSystemLoaderget_source`):

```
LOOKING FOR page.html in [u'/Users/bruce/Development/dylan/opendylan/documentation/library-reference/source/_templates']
LOOKING FOR page.html in [u'/Users/bruce/Development/dylan/opendylan/documentation/library-reference/source/../../_themes/opendylan-docs']
LOOKING FOR page.html in [u'/Users/bruce/Development/dylan/opendylan/documentation/library-reference/source/../../_themes/sphinx_rtd_theme']
LOOKING FOR page.html in ['/usr/local/lib/python2.7/site-packages/sphinx/themes/basic']
LOOKING FOR layout.html in [u'/Users/bruce/Development/dylan/opendylan/documentation/library-reference/source/_templates']
LOOKING FOR layout.html in [u'/Users/bruce/Development/dylan/opendylan/documentation/library-reference/source/../../_themes/opendylan-docs']
LOOKING FOR sphinx_rtd_theme/layout.html in [u'/Users/bruce/Development/dylan/opendylan/documentation/library-reference/source/_templates']
LOOKING FOR sphinx_rtd_theme/layout.html in [u'/Users/bruce/Development/dylan/opendylan/documentation/library-reference/source/../../_themes/opendylan-docs']
LOOKING FOR sphinx_rtd_theme/layout.html in [u'/Users/bruce/Development/dylan/opendylan/documentation/library-reference/source/../../_themes/sphinx_rtd_theme']
LOOKING FOR sphinx_rtd_theme/layout.html in ['/usr/local/lib/python2.7/site-packages/sphinx/themes/basic']
LOOKING FOR sphinx_rtd_theme/layout.html in ['../../_themes']
LOOKING FOR sphinx_rtd_theme/layout.html in ['/usr/local/lib/python2.7/site-packages/sphinx/themes']
```

**Comment by waywardmonkeys** [comment]
If I make my `html_theme_path` into an absolute path, then things work fine ... but the `conf.py` template says it can / should be relative.

**Comment by waywardmonkeys** [comment]
For now, I'm working around this by just setting my `html_theme_path` to `[os.path.join(os.path.dirname(__file__), '../../_themes')]`.

**Comment by shimizukawa** [comment]
Thanks for the details.

> `html_theme_path` to `[os.path.join(os.path.dirname(__file__), '../../_themes')]`.

`[os.path.abspath('../../_themes')]` is also fine.

> If I make my html_theme_path into an absolute path, then things work fine ... but the conf.py template says it can / should be relative.

conf.py saied:

```
# Add any paths that contain custom themes here, relative to this directory.
```

It means, "If you specified relative path, ..." then absolute path you specified is not wrong.
However, the relative path `'../../_themes'` should be fine as `conf.py` mentioned.
I guess it's a sphinx bug.

**Comment by shimizukawa** [comment]
Maybe this is fixed by 9d0ce7a.

**Comment by shimizukawa** [comment]
https://dl.dropboxusercontent.com/u/284189/sphinx-issues/issue1792_2.tgz can't be built by Sphinx-1.3.1.
Now sphinx 1.3.2 (after 9d0ce7a, not released) can build it.
